packages/node_modules/pouchdb-core/src/adapter.js

```diff
@@ -217,10 +218,43 @@ AbstractPouchDB.prototype.put = adapterFun('put', function (doc, opts, cb) {
       return this._putLocal(doc, cb);
     }
   }
-  if (typeof this._put === 'function' && opts.new_edits !== false) {
-    this._put(doc, opts, cb);
+  const self = this;
+  if (opts.force && doc._rev) {
+    transformForceOptionToNewEditsOption(function () {
+      putDoc(function (err) {
+        const result = err ? null : {ok: true, id: doc._id, rev: doc._rev};
+        cb(err, result);
+      });
+    });
   } else {
-    this.bulkDocs({docs: [doc]}, opts, yankError(cb));
+    putDoc(cb);
+  }
+
+  function transformForceOptionToNewEditsOption(next) {
+    self.get(doc._id, {revs: true, rev: doc._rev}, function (err, obj) {
+      /* istanbul ignore if */
+      if (err) {
+        return next(err);
+      }
+      const revisions = obj._revisions;
+      const nRevNum = revisions.start + 1;
+      const newRevId = uuid(32, 16).toLowerCase();
+      revisions.start = nRevNum;
+      revisions.ids.unshift(newRevId);
```

Comment:
Sorry, I am not familiar with the codebase.
you mean it is ok to have only partial revisions history in `doc._revisions.ids` when `bulkDocs` with `new_edits=false`? 
If so, your proposed implementation is definitely better, and I will adopt it.